test/irmin-bench/ema.ml

```diff
@@ -0,0 +1,191 @@
+  let ema = feed ~n:99 42. ema |> feed_none ~n:1 in
+  Alcotest.(check bool) "1U, 1F, 100U, 1F" (Ema.is_relevant ema) true;
+  let ema = feed_none ~n:1 ema in
+  Alcotest.(check bool) "1U, 1F, 100U, 2F" (Ema.is_relevant ema) false;
+  Lwt.return_unit
+
+let test_relevance_low _ () =
+  let ema = Ema.create ~relevance_threshold:0.01 0.5 in
+  Alcotest.(check bool) "Ema empty" (Ema.is_relevant ema) false;
+  let ema = feed ~n:5 42. ema in
+  Alcotest.(check bool) "5U" (Ema.is_relevant ema) false;
+  let ema = feed ~n:10 42. ema in
+  Alcotest.(check bool) "10U" (Ema.is_relevant ema) true;
+  let ema = feed_none ~n:1 ema in
+  Alcotest.(check bool) "10U, 1F" (Ema.is_relevant ema) false;
+  Lwt.return_unit
+
+let test_relevance_one _ () =
+  let ema = Ema.create ~relevance_threshold:1.0 0.5 in
+  Alcotest.(check bool) "Ema empty" (Ema.is_relevant ema) false;
+  let ema = feed ~n:1 42. ema in
+  Alcotest.(check bool) "1U" (Ema.is_relevant ema) true;
+  let ema = feed_none ~n:10 ema in
+  Alcotest.(check bool) "1U, 10F" (Ema.is_relevant ema) true;
+  Lwt.return_unit
+
+let test_commutativity_shift _ () =
+  let add a b = List.map2 ( +. ) a b in
+  let mom = 0.1 in
+  let xs = [ 10.0; -5.0; 6.0; -7.0; -20.0; 5.0; 9.0 ] in
+  let ys = [ 7.0; 3.0; -6.0; -1.0; -3.0; 4.0; 7.0 ] in
+  let res = Ema.map mom (add xs ys) in
+  let res' = add (Ema.map mom xs) (Ema.map mom ys) in
+  Alcotest.(check (list (float 0.0001))) "ema(a + b) = ema(a) + ema(b)" res res';
+  Lwt.return_unit
+
+let test_commutativity_log_mul _ () =
+  let mul a b = List.map2 ( *. ) a b in
+  let log l = List.map Float.log l in
+  let exp l = List.map Float.exp l in
+  let mom = 0.1 in
+  let xs = [ 10.0; 5.0; 6.0; 7.0; 20.0; 5.0; 9.0 ] in
+  let ys = [ 7.0; 3.0; 6.0; 1.0; 3.0; 4.0; 7.0 ] in
+  let res = mul xs ys |> log |> Ema.map mom |> exp in
+  let res' =
+    mul (xs |> log |> Ema.map mom |> exp) (ys |> log |> Ema.map mom |> exp)
+  in
+  Alcotest.(check (list (float 0.0001)))
+    "exp(ema(log(a))) * exp(ema(log(b))) = exp(ema(log(a * b)))" res res';
+  Lwt.return_unit
+
+let test_cases =
+  [
+    ( "ema",
+      [
+        Alcotest_lwt.test_case "momentum" `Quick test_momentum;
+        Alcotest_lwt.test_case "momentum_zero" `Quick test_momentum_zero;
+        Alcotest_lwt.test_case "momentum_high" `Quick test_momentum_high;
+        Alcotest_lwt.test_case "from_half_life" `Quick test_from_half_life;
+        Alcotest_lwt.test_case "from_half_life_ratio" `Quick
+          test_from_half_life_ratio;
+        Alcotest_lwt.test_case "batch" `Quick test_update_batch;
+        Alcotest_lwt.test_case "relevance" `Quick test_relevance;
+        Alcotest_lwt.test_case "relevance_low" `Quick test_relevance_low;
+        Alcotest_lwt.test_case "relevance_one" `Quick test_relevance_one;
+        Alcotest_lwt.test_case "commutativity shift" `Quick
+          test_commutativity_shift;
+        Alcotest_lwt.test_case "commutativity logmul" `Quick
+          test_commutativity_log_mul;
```

Comment:
I think tests are unlikely to ever use `Lwt` (and this just generates noise). Can you switch them to use either `Alcotest.test_case` or `Alcotest_lwt.test_case_sync`?

Reply:
Sure